test/MaeParserTest.cpp

```diff
@@ -442,7 +441,13 @@ BOOST_AUTO_TEST_CASE(Integer)
     {
         std::stringstream ss("-2147483648");
         Buffer b(ss);
+#ifdef _MSC_VER
+        // MSVC is buggy:
+        // https://developercommunity.visualstudio.com/content/problem/141813/-2147483648-c4146-error.html
+        BOOST_REQUIRE_EQUAL(parse_value<int>(b), 0x80000000);
```

Comment:
I'm not sure that it is crucial that this be at the negative limit - maybe you could do: `-2147483647`. Or if it _is_ important to check the limit, maybe we should do that by construction - 

    const int reference = std::numeric_limits<int>::min();
    std::stringstream ss;
    ss << reference;
    Buffer b(ss)
    BOOST_REQUIRE_EQUAL(parse_value<int>(b), reference);

Reply:
Using the numeric_limits, good idea. But we only need to construct the int value, not the string.

Reply:
I was thinking that technically, the limit value is implementation defined. If someone compiles this on a 32bit machine (or a 128bit, I suppose...) the test would fail.

Reply:
I noticed right after writing: it's not likely but, depending on the system, the int might be different than 4 bytes. So we do need to update the string.